Source/Core/RequestScheduler.js

```diff
@@ -0,0 +1,256 @@
+/*global define*/
+define([
+        './Check',
+        './defined',
+        './defineProperties',
+        './Heap',
+        './isBlobUri',
+        './isDataUri',
+        './RequestState',
+        '../ThirdParty/when'
+    ], function(
+        Check,
+        defined,
+        defineProperties,
+        Heap,
+        isBlobUri,
+        isDataUri,
+        RequestState,
+        when) {
+    'use strict';
+
+    /**
+     * Tracks the number of active requests and prioritizes incoming requests.
+     *
+     * @exports RequestScheduler
+     *
+     * @private
+     */
+    function RequestScheduler() {
+    }
+
+    /**
+     * The maximum number of simultaneous active requests. Un-throttled requests do not observe this limit.
+     * @type {Number}
+     * @default 50
+     */
+    RequestScheduler.maximumRequests = 50;
+
+    /**
+     * The maximum size of the priority heap. This limits the number of requests that are sorted by priority. Only applies to requests that are not yet active.
+     * @type {Number}
+     * @default 20
+     */
+    RequestScheduler.priorityHeapSize = 20;
+
+    /**
+     * Specifies if the request scheduler should throttle incoming requests, or let the browser queue requests under its control.
+     * @type {Boolean}
+     * @default true
+     */
+    RequestScheduler.throttle = false;
+
+    /**
+     * When true, log statistics to the console every frame
+     * @type {Boolean}
+     * @default false
+     */
+    RequestScheduler.debugShowStatistics = false;
+
+    function sortRequests(a, b) {
+        // Sort requests by higher screen space error and then closer distance
+        if (a.screenSpaceError !== b.screenSpaceError) {
```

Comment:
Should the constructor function have any validation for making SSE/distance either/or?

Are we sure that this is even the right metric?  Maybe distance and how center the tile is in the viewport is more important than SSE, which would include distant tiles with large error.  Or maybe this is biased to distance/centeredness, but let's through `n` lower priority requests every so often to prevent stalling them but keep the most important detail coming in.